code/modules/economy/persistent_bank.dm

```diff
@@ -8,75 +8,109 @@
-			usr.client.set_last_purchase(null)
-			boutput( usr, "<span class='notice'><b>The thing you previously purchased has been removed from your inventory due to it no longer existing.</b></span>")
-
-		var/ret = "<p style=\"font-size:125%;\">BALANCE :  <b>[usr.client.persistent_bank]</b></p><br/>"
-		ret += "<p style=\"font-size:110%;\">HELD ITEM :  <b>[usr.client.persistent_bank_item ? usr.client.persistent_bank_item : "Nothing!"]</b></p><br/>"
-		ret += "Purchase an item for the upcoming round. Earn more cash by completing rounds.<br/>"
-		ret += "A purchased item will persist until you die or fail to escape the station. If you have a Held Item, buying a new one will replace it.<br/><br/>"
-		for(var/i=1, i <= persistent_bank_purchaseables.len, i++)
-			var/datum/bank_purchaseable/p = persistent_bank_purchaseables[i]
-
-			if(!p.hasJobXP(usr.client.key)) continue
-
-			ret += "[theme.generateButton(i, "[p.name] $[p.cost]")] <br/>"
-
-			//Convert the HELD ITEM string into a purchased item on Mind if applicable here
-			if (p.name == usr.client.persistent_bank_item)
-				var/mob/new_player/playermob = usr.client.mob
+			if(!p.hasJobXP(user.client.key)) continue
+			purchasables += list(
+				list(
+					"pname" = p.name,
+					"cost" = p.cost,
+					"img" = icon2base64(icon(initial(p.icon), initial(p.icon_state), dir=p.icon_dir, frame=p.icon_frame, moving=0)),
+				)
+			)
+			//If you have a purchased item: set held
+			// or you have a persistent item: set it as purchased, and set held
+			var/datum/bank_purchaseable/already_purchased = user.client?.mob?.mind?.purchased_bank_item
+			if (istype(already_purchased) && p.name == already_purchased.name)
+				found_held = p.name
+			else if (p.name == user.client.persistent_bank_item)
+				found_held = p.name
+				var/mob/new_player/playermob = user.client.mob
 				if (playermob.mind)
 					playermob.mind.purchased_bank_item = p
 
-		ret += "<br>This menu is a WIP! Expect additional items and other changes to come in the future!<br/>"
-		return ret
+		if(user.client.persistent_bank_item && user.client.persistent_bank_item != "none" && !found_held)
+			user.client.set_last_purchase(null)
+			boutput(user, "<span class='notice'><b>The thing you previously purchased has been removed from your inventory due to it no longer existing.</b></span>")
+
+		var/truebalance = user.client.persistent_bank
+		if(istype(src.bought_this_round))
+			truebalance += src.bought_this_round.cost
+		. = list(
+			"purchasables" = purchasables,
+			"held" = found_held,
+			"balance" = user.client.persistent_bank,
+			"truebalance" = truebalance
+		)
+
+	ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+		. = ..()
+		if (.)
+			return
+		if(current_state >= GAME_STATE_PLAYING)
+			boutput(ui.user, "<span class='notice'><b>The round has started, you'll have to wait until the next round!</b></span>" )
+			ui.close()
+			return
```

Comment:
I think this might prevent people who join mid-round from making Spacebux purchases?

I was thinking more a case of seeing if the person using it had spawned? Basically, mind verifying this works as expected if I open the Spacebux menu, spawn in, then try to use it 30 seconds later?

Reply:
oh dam u rite

hmm, need a check for if the player has spawned, rather than game has started then